src/backend/distributed/utils/distribution_column.c

```diff
@@ -115,6 +116,38 @@ column_to_column_name(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * FindColumnWithNameOnTargetTable gets a source table and column. The function
+ * returns the the column with the same name on the target table.
+ *
+ * Note that due to dropping columns, the parent's distribution key may not
+ * match the partition's distribution key. See issue #5123.
+ *
+ * The function throws error if the input or output is not valid.
+ */
+Var *
+FindColumnWithNameOnTargetTable(Oid sourceRelationId, Var *sourceColumn,
+								Oid targetRelationId)
+{
+	if (sourceColumn == NULL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_UNDEFINED_COLUMN),
+						errmsg("cannot find the given column on table \"%s\"",
+							   generate_qualified_relation_name(sourceRelationId))));
+	}
+
+	char *sourceDistributionKeyStr = nodeToString(sourceColumn);
+	char *columnName = ColumnToColumnName(sourceRelationId, sourceDistributionKeyStr);
+	AttrNumber attributeNumberOnTarget = get_attnum(targetRelationId, columnName);
+	Var *targetColumn =
+		makeVar(sourceColumn->varno, attributeNumberOnTarget,
+				sourceColumn->vartype, sourceColumn->vartypmod,
+				sourceColumn->varcollid, sourceColumn->varlevelsup);
```

Comment:
I think I like the function returns `Var` so that we can put the details of creating the `Var` in to the function.

I also agree with you to make it more generic. What do you think about the new API?